ttnn/cpp/ttnn/operations/data_movement/pad/pad.hpp

```diff
@@ -5,361 +5,41 @@
-            }
-            else if(original_rank == 2) {
-                output_tensor =  pad_impl<tt::tt_metal::Array2D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 3) {
-                output_tensor =  pad_impl<tt::tt_metal::Array3D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 4) {
-                output_tensor =  pad_impl<tt::tt_metal::Array4D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 5) {
-                output_tensor =  pad_impl<tt::tt_metal::Array5D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 6) {
-                output_tensor =  pad_impl<tt::tt_metal::Array6D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 7) {
-                output_tensor =  pad_impl<tt::tt_metal::Array7D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-            else if(original_rank == 8) {
-                output_tensor =  pad_impl<tt::tt_metal::Array8D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-            }
-        }
-        else {
-            output_tensor =  pad_impl<tt::tt_metal::Array4D>(queue_id, input_tensor, padding, value, use_multicore, memory_config_arg);
-        }
-        // output_tensor is currently 4D. We have to squeeze back to the original rank
-        auto to_vec = [](const auto& arr) {return std::vector<uint32_t>(arr.begin(), arr.end());};
-        auto shape = to_vec(output_tensor.get_shape().value);
-        auto padded_shape = to_vec(output_tensor.get_shape().with_tile_padding().value);
-        if (auto rank_diff = shape.size() - original_rank; rank_diff) {
-            auto remove_first_elements = [](auto& source, size_t n) {
-                source.erase(source.begin(), source.begin() + n);
-            };
-            remove_first_elements(shape, rank_diff);
-            remove_first_elements(padded_shape, rank_diff);
-            auto squeezedShape = ttnn::Shape(tt::tt_metal::Shape(shape, padded_shape));
-            output_tensor = ttnn::reshape(output_tensor, squeezedShape);
-        }
-
-        // Padding always turns the intended shape to the shape with tile padding. For simplicity of the operation
-        output_tensor = ttnn::reshape(output_tensor, ttnn::Shape(padded_shape));
-
-        return output_tensor;
-    }
+    static ttnn::Tensor operator()(uint8_t queue_id,
+                                   const ttnn::Tensor& input_tensor,
+                                   std::vector<std::pair<uint32_t, uint32_t>> padding,
```

Comment:
could you pass vector by reference or try std::span to check if it compiles?

Reply:
done. But won't touch impl, it modifies padding